dspace-api/src/main/java/org/dspace/administer/CreateAdministrator.java

```diff
@@ -114,91 +144,124 @@ protected void negotiateAdministratorDetails()
         String email = null;
         String firstName = null;
         String lastName = null;
-        char[] password1 = null;
-        char[] password2 = null;
         String language = I18nUtil.getDefaultLocale().getLanguage();
+        ConfigurationService cfg = DSpaceServicesFactory.getInstance().getConfigurationService();
+        boolean flag = false;
+        char[] password = null;
+
+
+        if (!line.hasOption('p') && line.hasOption("e")  && line.hasOption("f") && line.hasOption("l")
+            && (line.hasOption("c") || (!line.hasOption("c")
+            && cfg.getProperty("webui.supported.locales") == null))) {
+            flag = true;
+            dataOK = false;
+        } else if (!line.hasOption('p') && !line.hasOption("e")  && !line.hasOption("f") && !line.hasOption("l")) {
+            flag = false;
+            dataOK = false;
+        } else if (line.hasOption('p') && line.hasOption("e")  && line.hasOption("f") && line.hasOption("l")
+                   && (line.hasOption("c") || (!line.hasOption("c")
+                   && cfg.getProperty("webui.supported.locales") == null))) {
+            dataOK = true;
+            password = line.getOptionValue("p").toCharArray();
+        }
 
         while (!dataOK) {
-            System.out.print("E-mail address: ");
-            System.out.flush();
-
-            email = console.readLine();
-            if (!StringUtils.isBlank(email)) {
-                email = email.trim();
-            } else {
-                System.out.println("Please provide an email address.");
-                continue;
-            }
-
-            System.out.print("First name: ");
-            System.out.flush();
-
-            firstName = console.readLine();
+            if (flag) {
+                password = getPassword(console);
```

Comment:
You might want to pull the password prompt up out of the loop.  The user has no way to check whether the password he typed is the one that he meant, so there's no way to correct it.  The password has a different check:  was it entered the same way twice?

Reply:
will look into this